### PR TITLE
fix serializer for enums

### PR DESCRIPTION
**Pull request by roaminro**
Calling `this.root.lookupType(type);` on an `enum` type throws an error. The solution in this PR is to not decode the data if the type doesn't exists instead of throwing the error.

Another solution would be to use `this.root.lookupEnum(type);` when `this.root.lookupType(type);` fails, however, we need to decide how we want to present the `enum` value to the library's users (integer representation vs string representation)(this current PR will just return the integer representation of the enum value)

### CODE REVIEW

**Comment by joticajulian** [comment]
I don't recommend to add a try catch because the dev will not be able to spot errors in the code. I saw we can use `this.root.lookTypeOrEnum()` instead to see if it is an enum.

The decision to represent the `enum` as number or string should not be done in `btypeEncode` / `btypeDecode`, because these functions only modify the representation of `bytes`, which is a particular case in the format representation used in koinos (address, base64, hex, base58, etc).
Instead of that we could use built-in functions in [protobufjs](https://www.npmjs.com/package/protobufjs) lib.

Here is an example from their documentation which is used to represent the enums as string:
```
var object = AwesomeMessage.toObject(message, {
  enums: String,  // enums as string names
  longs: String,  // longs as strings (requires long.js)
  bytes: String,  // bytes as base64 encoded strings
  defaults: true, // includes default values
  arrays: true,   // populates empty arrays (repeated fields) even if defaults=false
  objects: true,  // populates empty objects (map fields) even if defaults=false
  oneofs: true    // includes virtual oneof fields set to the present field's name
});
```
But by default they are numbers... so maybe we could add an option in the serializer class to specify this part and let the dev decide the preferred option. But for the moment we can continue with the default option (numbers).

So, in conclusion, I propose to use `this.root.lookTypeOrEnum()` and return the item without modifications if it is an enum.

**Comment by joticajulian** [comment]
I fixed it here https://github.com/joticajulian/koilib/commit/067cf02df26a1f0f0c236dba1d1b300873a927bf
It is also published in version 5.1.1
